docs/proposals/backend-support-appset.md

```diff
@@ -0,0 +1,226 @@
+
+6. Once the pre-checks have been confirmed, the controller will run the generator, and render the parameters into the template. Upon generating the template, the controller will need to perform some checks before creating the Applications.
+
+7. ***Post-generate check:*** Look at the generated Applications (but don't apply them yet!), and verify that user has the required RBAC permissions to perform the required actions. This is a dynamic (or runtime) check, as it works on the dynamically generated applications; eg. it is not possible to predict the result of these checks without first running the generator, unlike the static checks.
+
+8. Once all the checks have passed, apply the ApplicationSet and the Applications, to the namespace.
+
+#### **Should we add a new 'applicationset' RBAC resource?**
+
+* Do we need to add a new RBAC resource for applicationsets, alongside the existing ones? (clusters, projects, applications, repositories, certificates, accounts, gpgkeys)
+Not mandatory for the design explained in this proposal.
+
+* If we DID add a new RBAC resource, this would require application administrators to add a new ApplicationSet resource to the RBAC policy list, alongside their existing Application policies and to keep the ApplicationSet and Application RBAC policy lists in sync. Or, said another way: there would be very few (if any) cases where a user would need to be able to create an Application, but not be able to create an ApplicationSet (and vice versa). When it comes to security, the less moving parts that need to be synchronized, the better (with some minor loss in flexibility, eg the ability to specifically prevent access to ApplicationSet resource).
+
+* If there is no RBAC resource for applicationsets, how do we control access to them?
+Instead of an applicationset rbac resource, we instead examine the Applications that are owned by the applicationset. An ApplicationSet inherits permissions from its generated children.
+Example: if you try to delete an ApplicationSet, we check if you can delete it’s child Applications.
+
+#### **ApplicationSet Deletion Algorithm:**
+
+A user is only able to delete an ApplicationSet if they have permissions to delete all of the Applications managed by the ApplicationSet. This check is performed in ApplicationSet controller, on receiving a delete request via GRPC from API server.
+
+* For each application owned by the ApplicationSet that the user is attempting to delete:
+    * Check if the user has delete permission on the Application
+    * Check if the user has delete permission within the project (?)
+* If the user does NOT have permission on least one of these, the operation should fail.
+* On pass, ApplicationController server deletes (ie kubectl delete) the ApplicationSet resource.
+
+#### **ApplicationSet Update Algorithm:**
+
+A user can only update an ApplicationSet if the user has permission to update all of the Applications currently owned by the ApplicationSet.
```

Comment:
I think a user should only be allowed to update an applicationset if they are allowed to create and delete and update all applications which can theoretically be owned by an applicationset. Because if someone can update an applicationset they can create and update and delete all applications which are manageable by that appset.

Reply:
Same comment contents as above, it would appear.